xcom2-launcher/xcom2-launcher/Classes/Mod/ModList.cs

```diff
@@ -279,13 +280,12 @@ internal void UpdateMod(ModEntry m, Settings settings)
                 try
                 {
                     // Parse .XComMod file
-                    var modinfo = new ModInfo(m.GetModInfoFile());
-                    m.SetRequiresWOTC(modinfo.RequiresXPACK);
-
+                    var modInfo = new ModInfo(m.GetModInfoFile());
+                    m.SetRequiresWOTC(modInfo.RequiresXPACK);
                 }
-                catch (InvalidOperationException)
+                catch (InvalidOperationException ex)
                 {
-                    return;
+                    Debug.WriteLine(ex.Message);
```

Comment:
What will this do on "live" builds?

Reply:
Debug calls are removed on a Release build (DEBUG not defined). A debug message is better that just ignoring the result when debugging. The "return" was redundant because the execution path in that function is finished at that point.

Reply:
I see your logic. Maybe it's better to throw an exception inside `#if DEBUG`? Messages in "output" window can be easily missed

Reply:
An InvalidOperationException maybe expected at that point. It is generally acceptable to swallow specific exception that are expected to happen but can't be or aren't required to be handled. However, it is always good to leave a code comment in that case. Now we don't what exactly the original author intended, but I think we can leave it this way. I already thought about implementing a "Logger" for cases like this. Its really nice to have a log file to be able to follow program flow in case of issues.